R/utils.R

```diff
@@ -100,7 +104,7 @@ maybe_n <- function(x) {
 }
 
 is_testing <- function() {
-  identical(Sys.getenv("TESTTHAT"), "true")  
+  requireNamespace("testthat", quietly = TRUE) && identical(Sys.getenv("TESTTHAT"), "true")
```

Comment:
That check shouldn't be needed since only testthat should be setting the envvar.

Reply:
It should because later on there are some function calling testthat function inside the package (skip_*). I swapped the order though so that if the variable is not present the package won't be loaded.

Reply:
I don't think it's a good idea to change `is_testing()` because it's a standard helper you'd generally expect to be copied directly from testthat. Where is `skip()` being used?

Reply:
There are two cases where this happens; on http.R lines 309-312:

```r
handle_rate_limit <- function(x, api, retryonratelimit = NULL, verbose = TRUE) {
  if (is_testing()) {
    testthat::skip("Rate limit exceeded")
  }
....
```
and auth.R lines 273-275 
```r
no_token <- function() {
  if (is_testing()) {
    testthat::skip("Auth not available")
...
```

I could move the check outside the function, but I thought it was more accurately to leave it inside it.